### PR TITLE
*: added builtin function sqrt

### PR DESCRIPTION
**Pull request by zhexuany**
- [x] sqrt function is implemented in this PR
- [x] test case is also added in this PR. 
- [x]  passed all test cases. 
A function wanted by #2420.

### CODE REVIEW

File: expression/builtin_math_test.go [hunk 1]
```diff
@@ -258,3 +258,25 @@ func (s *testEvaluatorSuite) TestSign(c *C) {
...
+	defer testleak.AfterTest(c)()
+	tbl := []struct {
+		Arg interface{}
+		Ret interface{}
+	}{
+		{nil, nil},
+		{int64(1), float64(1)},
+		{float64(4), float64(2)},
+		{"4", float64(2)},
+		{"9", float64(3)},
```
Comment (shenli) [inline expression/builtin_math_test.go]: Please add a test case for mysql> SELECT SQRT(-16);
        -> NULL

Comment (zhexuany) [inline expression/builtin_math_test.go]: DONE

**Comment by shenli** [comment]
LGTM
@XuHuaiyu PTAL

**Comment by zhexuany** [comment]
Just rebase current br onto master. All master changes are in this br, shall be ready to merge.

File: expression/builtin_math.go [hunk 2]
```diff
@@ -552,6 +552,49 @@ func builtinSign(args []types.Datum, ctx context.Context) (d types.Datum, err er
 	return d, nil
 }
 
+type sqrtFunctionClass struct {
```
Comment (XuHuaiyu) [inline expression/builtin_math.go]: Please add a definition of builtinFunctionClass and builtinSprtSig at the beginning of this file.

File: expression/builtin_math.go [hunk 3]
```diff
@@ -552,6 +552,49 @@ func builtinSign(args []types.Datum, ctx context.Context) (d types.Datum, err er
...
+		return args[0], nil
+	}
+
+	sc := ctx.GetSessionVars().StmtCtx
+	f, err := args[0].ToFloat64(sc)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+
+	// negative value does not have any square root in rational number
```
Comment (XuHuaiyu) [inline expression/builtin_math.go]: Add `.` at the end of this comment.

File: expression/builtin_math.go [hunk 4]
```diff
@@ -552,6 +552,49 @@ func builtinSign(args []types.Datum, ctx context.Context) (d types.Datum, err er
...
+	// Need return null directly
```
Comment (XuHuaiyu) [inline expression/builtin_math.go]: ditto.

Comment (zhexuany) [inline expression/builtin_math.go]: DONE

File: expression/builtin_math.go (hunk 3 above)
Comment (zhexuany) [inline expression/builtin_math.go]: DONE

File: expression/builtin_math.go (hunk 2 above)
Comment (zhexuany) [inline expression/builtin_math.go]: DONE

**Comment by shenli** [comment]
@XuHuaiyu PTAL

**Comment by XuHuaiyu** [comment]
LGTM

**Pull request merged**